src/main/ruby/rack/handler/servlet/servlet_env.rb

```diff
@@ -112,9 +112,10 @@ def [](key)
         #
         # @param key the param name
         # @param val the value(s) in a array-like structure
-        # @param hash the Hash to store the name, value pair
-        def store_parameter(key, val, hash)
+        # @param params the Hash to store the name, value pair
+        def store_parameter2(params, key, val, depth = 0)
           # emulating Rack::Utils.parse_nested_query behavior
+          raise ::Rack::Utils::ParamsTooDeepError if depth >= 32
```

Comment:
aren't these configurable in Rack itself?

Reply:
It seems just a hard coded security thing :(